ios/Classes/NativeBridge/Handlers/MobileAuthHandler.swift

```diff
@@ -2,132 +2,128 @@ import Foundation
 import OneginiSDKiOS
 
 protocol MobileAuthConnectorToHandlerProtocol: AnyObject {
-    func enrollForMobileAuth(_ completion: @escaping (Bool?, SdkError?) -> Void)
-    func isUserEnrolledForMobileAuth() -> Bool
-    func handleMobileAuthConfirmation(cancelled: Bool)
-    func handleOTPMobileAuth(_ otp: String , customRegistrationChallenge: ONGCustomRegistrationChallenge?, _ completion: @escaping (Any?, SdkError?) -> Void)
-    func handleQrOTPMobileAuth(_ otp: String , customRegistrationChallenge: ONGCustomRegistrationChallenge?, _ completion: @escaping (Any?, SdkError?) -> Void)
-}
-
-enum MobileAuthAuthenticatorType: String {
-    case fingerprint = "biometric"
-    case pin = "PIN"
-    case confirmation = ""
+    func handleMobileAuthWithOtp(otp: String, completion: @escaping (Result<Void, FlutterError>) -> Void)
+    func enrollMobileAuthentication(completion: @escaping (Result<Void, FlutterError>) -> Void)
+    func acceptMobileAuthRequest(completion: @escaping (Result<Void, FlutterError>) -> Void)
+    func denyMobileAuthRequest(completion: @escaping (Result<Void, FlutterError>) -> Void)
+    func setMobileAuthCallback(didReceiveConfirmationChallenge confirmation: @escaping (Bool) -> Void)
```

Comment:
You don't need `didReceiveConfirmationChallenge`, just  `func setMobileAuthCallback(_ confirmation: @escaping (Bool) -> Void)`. 
BTW
1. why do you need extra "callback" if every method has completion?
2. why bool insted of some result?

Reply:
1. the mobileAuthCallback is the one that we need to accept or deny during the mobileauthentication flow; while the other completions are the completions for the functions themselves 
2. bool is the one we receive from the native SDK 
3. replaced the `didReceiveConfirmationChallenge` with _